controllers/user/controllers/account_controller.go

```diff
@@ -47,9 +55,61 @@ type AccountReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
 func (r *AccountReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	payment := &userv1.Payment{}
+	if err := r.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: req.Name}, payment); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	if payment.Spec.UserID == "" || payment.Spec.Amount == 0 || payment.Status.TradeNO == "" {
+		return ctrl.Result{}, fmt.Errorf("payment is invalid: %v", payment)
+	}
+	if payment.Status.Status == pay.StatusSuccess {
+		return ctrl.Result{}, nil
+	}
+
+	account := &userv1.Account{}
+	account.Name = payment.Spec.UserID
+	err := r.Get(ctx, req.NamespacedName, account)
+	if errors.IsNotFound(err) {
+		account.Name = payment.Spec.UserID
+		account.Status.Balance = 0
+		account.Status.ChargeList = []userv1.Charge{}
+		if err := r.Create(ctx, account); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
 
-	// TODO(user): your logic here
+	status, err := pay.QueryOrder(payment.Status.TradeNO)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	switch status {
+	case pay.StatusSuccess:
+		account.Status.ChargeList = append(account.Status.ChargeList, userv1.Charge{
+			Amount:  payment.Spec.Amount,
+			Time:    metav1.Now(),
+			Status:  status,
+			TradeNO: payment.Status.TradeNO,
+		})
+		account.Status.Balance += payment.Spec.Amount
+		if err := r.Status().Update(ctx, account); err != nil {
+			return ctrl.Result{}, err
+		}
+		payment.Status.Status = status
+		if err := r.Status().Update(ctx, payment); err != nil {
+			return ctrl.Result{}, err
+		}
+	case pay.StatusProcessing:
```

Comment:
what is `processing` status mean?

Reply:
Paying, users scanning the qrcode after placing an order